# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: architecture/sparc64/pkg_gcc.diff
# Copyright (C) 2004 - 2015 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---

Default to a 32-bit user-land - mostly (still) for performance these days.

	- Rene Rebe <rene@exactcode.de>

--- gcc-4.9.2/gcc/config/sparc/linux64.h.vanilla	2015-02-15 14:56:58.340000000 +0000
+++ gcc-4.9.2/gcc/config/sparc/linux64.h	2015-02-15 14:58:34.392000000 +0000
@@ -32,12 +32,14 @@
 
 /* On Linux, the combination sparc64-* --with-cpu=v8 is supported and
    selects a 32-bit compiler.  */
+#ifdef SPARC64_BECAME_ROCK_SOLID_AND_X_USABLE_RENER
 #if defined(TARGET_64BIT_DEFAULT) && TARGET_CPU_DEFAULT >= TARGET_CPU_v9
 #undef TARGET_DEFAULT
 #define TARGET_DEFAULT \
   (MASK_V9 + MASK_PTR64 + MASK_64BIT + MASK_STACK_BIAS + \
    MASK_APP_REGS + MASK_FPU + MASK_LONG_DOUBLE_128)
 #endif
+#endif
 
 /* This must be v9a not just v9 because by default we enable
    -mvis.  */
